### ISSUE TITLE
Error in Go to Super Method

### ISSUE THREAD

**Issue by DanTup** [Dart-Code/Dart-Code#815]
Working on `flutter_tools` on a new file with a class that inherited `PollingDeviceDiscovery`:

```dart
class AndroidAvds extends PollingDeviceDiscovery {
  AndroidAvds() : super('Android emulators');

  @override
  bool get supportsPlatform => true;

  @override
  bool get canListAnything => getEmulatorPath(androidSdk) != null;

  @override
  Future<List<Device>> pollingGetDevices() async => getEmulatorAvds();
}
```

Invoked it on `canListAnything` and got this error:

<img width="506" alt="screen shot 2018-04-18 at 09 30 02" src="https://user-images.githubusercontent.com/1078012/38920462-39ef7bf2-42eb-11e8-99af-fa8d81ca8757.png">

**Comment by DanTup** [comment]
```js
[
	{
		"id": "23",
		"method": "analysis.getNavigation",
		"params": {
			"file": "/Users/dantup/Dev/Google/flutter/packages/flutter_tools/lib/src/android/android_avd.dart",
			"length": 0,
			"offset": 533
		}
	},
	{
		"id": "23",
		"result": {
			"files": [
				"/Users/dantup/Dev/Google/flutter/packages/flutter_tools/lib/src/base/file_system.dart"
			],
			"targets": [
				{
					"kind": "LIBRARY",
					"fileIndex": 0,
					"offset": 0,
					"length": 0,
					"startLine": 1,
					"startColumn": 1
				}
			],
			"regions": [
				{
					"offset": 514,
					"length": 26,
					"targets": [
						0
					]
				}
			]
		}
	}
]
```

**Comment by DanTup** [comment]
Fixed in f34c565.
